### PR TITLE
Fix instance import

### PR DESCRIPTION
**Pull request by lukaszcz**
Fixes a bug which prevented some instances from being imported from other modules.

### CODE REVIEW

File: src/Juvix/Compiler/Internal/Data/InstanceInfo.hs [hunk 1]
```diff
@@ -29,20 +29,34 @@ data InstanceInfo = InstanceInfo
...
+  }
 
 makeLenses ''InstanceApp
 makeLenses ''InstanceInfo
+makeLenses ''InstanceTable
+
+instance Semigroup InstanceTable where
+  t1 <> t2 =
+    InstanceTable $
+      HashMap.unionWith (++) (t1 ^. instanceTableMap) (t2 ^. instanceTableMap)
```
Comment (janmasrovira) [inline src/Juvix/Compiler/Internal/Data/InstanceInfo.hs:43]: Should we handle deletion of duplicates or is it guaranteed that these two lists are disjoint?

File: src/Juvix/Compiler/Internal/Data/InstanceInfo.hs [hunk 2]
```diff
@@ -29,20 +29,34 @@ data InstanceInfo = InstanceInfo
...
-type InstanceTable = HashMap InductiveName [InstanceInfo]
+newtype InstanceTable = InstanceTable
+  { _instanceTableMap :: HashMap InductiveName [InstanceInfo]
+  }
 
 makeLenses ''InstanceApp
 makeLenses ''InstanceInfo
+makeLenses ''InstanceTable
+
+instance Semigroup InstanceTable where
```
Comment (janmasrovira) [inline src/Juvix/Compiler/Internal/Data/InstanceInfo.hs:40]: You can derive semigroup and monoid using `deriving newtype (Monoid, Semigroup)`.
I am not saying you should, just a remainder

File: src/Juvix/Compiler/Internal/Data/InstanceInfo.hs (hunk 1 above)
Comment (lukaszcz) [reply src/Juvix/Compiler/Internal/Data/InstanceInfo.hs:43]: I think they will be disjoint - this is invoked when we're combining things from different modules

File: src/Juvix/Compiler/Internal/Data/InstanceInfo.hs (hunk 2 above)
Comment (lukaszcz) [reply src/Juvix/Compiler/Internal/Data/InstanceInfo.hs:40]: But the whole point that is we want different behaviour for Semigroup?

File: src/Juvix/Compiler/Internal/Data/InstanceInfo.hs (hunk 1 above)
Comment (janmasrovira) [reply src/Juvix/Compiler/Internal/Data/InstanceInfo.hs:43]: If I open the same module twice I'm getting an ambiguity error.

`Main.juvix`
```
module Main;

import Default open;
import Default open;

x : Unit := default;
```

Default.jvuix
```
module Default;

type Unit := unit;

trait
type Default A := mkDefault { default : A };

instance
Unit-Default : Default Unit := mkDefault unit;

default {A : Type} {{x : Default A}} : A := case x
 | mkDefault a := a;
```

File: src/Juvix/Compiler/Internal/Data/InstanceInfo.hs (hunk 2 above)
Comment (janmasrovira) [reply src/Juvix/Compiler/Internal/Data/InstanceInfo.hs:40]: ah yes! The semigroup instance for map just replaces the contents

File: src/Juvix/Compiler/Internal/Data/InstanceInfo.hs (hunk 1 above)
Comment (lukaszcz) [reply src/Juvix/Compiler/Internal/Data/InstanceInfo.hs:43]: Ah, Ok, then I'll add filtering duplicates

Comment (lukaszcz) [reply src/Juvix/Compiler/Internal/Data/InstanceInfo.hs]: done

**Comment by lukaszcz** [comment]
This requires more work. It seems also something is wrong with the dependencyBuilder, because with imports some instance identifiers are wrongly filtered out.

File: src/Juvix/Compiler/Internal/Data/InstanceInfo.hs (hunk 1 above)
Comment (janmasrovira) [reply src/Juvix/Compiler/Internal/Data/InstanceInfo.hs]: did you push the changes? I can't see it in the diff

**Review by janmasrovira — APPROVED** [review]

**Pull request merged**